generator/src/test/java/com/scottlogic/deg/generator/generation/fieldvaluesources/datetime/SequentialDateIteratorTest.java

```diff
@@ -0,0 +1,123 @@
+package com.scottlogic.deg.generator.generation.fieldvaluesources.datetime;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+
+class SequentialDateIteratorTest {
+
+    private OffsetDateTime inclusiveMinDate;
+    private OffsetDateTime exclusiveMaxDate;
+    private OffsetDateTime referenceDate;
+
+
+    @BeforeEach
+    public void setup() {
+        // set some default values
+        this.inclusiveMinDate = OffsetDateTime.of(2001, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC);
+        this.exclusiveMaxDate= OffsetDateTime.of(2010, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC);
+    }
+
+    @Test
+    void checkHasNextWhenCurrentIsEqualToMax() {
```

Comment:
I think we do have a bit of a standard with unit tests. Arrange/Act/Assert and using methodName_withInputs_doesSomething as a test name.